internal/exec/stages/files/units.go

```diff
@@ -118,10 +120,11 @@ func (s *stage) createUnits(config types.Config) error {
 	}
 	// if we have presets then create the systemd preset file.
 	if len(presets) != 0 {
-		if err := s.relabelPath(filepath.Join(s.DestDir, util.PresetPath)); err != nil {
-			return err
-		}
-		if err := s.createSystemdPresetFile(presets); err != nil {
+		// useless as it will be done later in createSystemdPresetFiles
+		// if err := s.relabelPath(filepath.Join(s.DestDir, util.PresetPath)); err != nil {
```

Comment:
Good catch on the double relabel, but the remaining call site needs to call `relabelPath`, since that handles the parent directories as well.  Let's move this part of the change into a separate commit with a commit message explaining the reason for the change.

Reply:
I partially agree as I should modify this part according to multiple presets paths for global and system scope.
If I modify this part of code I will copy/paste what I've written in createSystemdPresetFiles
Am I missing something ?

Reply:
Discussed more OOB and I think we were talking past each other here.  The second call site now uses `relabelPath` so I'll resolve this.